### DISCUSSION TITLE
How to get the balance of Solana accoount?

### DISCUSSION THREAD

**Discussion by Chandra-Sekhar-Bala** [sol4k/sol4k#99] [Q&A] [answered]
How do I get the account balance of Solana by providing public address?

**Comment by Shpota** [comment, marked as answer]
You can get it using the `getBalance()` function.

```kotlin
val connection = Connection(rpcUrl)
val wallet = PublicKey("DxPv2QMA5cWR5Xfg7tXr5YtJ1EEStg5Kiag9HhkY1mSx")
val balance = connection.getBalance(wallet)
```

Please check [integration tests](https://github.com/sol4k/sol4k/blob/main/src/integration-test/kotlin/org/sol4k/ConnectionTest.kt) or [this repository](https://github.com/sol4k/sol4k-examples) to find ready-to-run examples.

**Comment by Chandra-Sekhar-Bala** [comment]
Thanks @Shpota, This works on getting account balance. 
But account details comes null. 

code: 
```
val publicKey: PublicKey = PublicKey(keypair.publicKey.toBase58())
val balanceResponse = connection.getBalance(publicKey)
val account = connection.getAccountInfo(keypair.publicKey)
logThis("Sol Balance Response: $balanceResponse")
logThis("Sol Account Response: $account")             
```

Logcat: 
```
Sol Balance Response: 0
Sol Account Response: null
```

**Reply by Shpota** [reply]
`getAccountInfo` returns null if account doesn't exist. You can find more details in [the getAccountInfo RPC documentation](https://solana.com/docs/rpc/http/getaccountinfo). Make sure the account exists on-chain and that you are using a proper network (dev/test/mainnet).
